packages/@aws-cdk/aws-sagemaker/lib/endpoint-config.ts

```diff
@@ -0,0 +1,313 @@
+   * @param id the resource id.
+   * @param endpointConfigName the name of the endpoint configuration.
+   */
+  public static fromEndpointConfigName(scope: Construct, id: string, endpointConfigName: string): IEndpointConfig {
+    const endpointConfigArn = cdk.Stack.of(scope).formatArn({
+      service: 'sagemaker',
+      resource: 'endpoint-config',
+      resourceName: endpointConfigName,
+    });
+
+    class Import extends cdk.Resource implements IEndpointConfig {
+      public endpointConfigArn = endpointConfigArn;
+      public endpointConfigName = endpointConfigName;
+    }
+
+    return new Import(scope, id, {
+      environmentFromArn: endpointConfigArn,
+    });
+  }
+
+  /**
+   * The ARN of the endpoint configuration.
+   */
+  public readonly endpointConfigArn: string;
+  /**
+   * The name of the endpoint configuration.
+   */
+  public readonly endpointConfigName: string;
+
+  private readonly _instanceProductionVariants: { [key: string]: InstanceProductionVariant; } = {};
+
+  constructor(scope: Construct, id: string, props: EndpointConfigProps = {}) {
+    super(scope, id, {
+      physicalName: props.endpointConfigName,
+    });
+
+    (props.instanceProductionVariants || []).map(p => this.addInstanceProductionVariant(p));
+
+    // create the endpoint configuration resource
+    const endpointConfig = new CfnEndpointConfig(this, 'EndpointConfig', {
+      kmsKeyId: (props.encryptionKey) ? props.encryptionKey.keyArn : undefined,
+      endpointConfigName: this.physicalName,
+      productionVariants: cdk.Lazy.any({ produce: () => this.renderInstanceProductionVariants() }),
+    });
+    this.endpointConfigName = this.getResourceNameAttribute(endpointConfig.attrEndpointConfigName);
+    this.endpointConfigArn = this.getResourceArnAttribute(endpointConfig.ref, {
+      service: 'sagemaker',
+      resource: 'endpoint-config',
+      resourceName: this.physicalName,
+    });
+  }
+
+  /**
+   * Add production variant to the endpoint configuration.
+   *
+   * @param props The properties of a production variant to add.
+   */
+  public addInstanceProductionVariant(props: InstanceProductionVariantProps): void {
+    if (props.variantName in this._instanceProductionVariants) {
+      throw new Error(`There is already a Production Variant with name '${props.variantName}'`);
+    }
+    this.validateProps(props);
+    this._instanceProductionVariants[props.variantName] = {
+      acceleratorType: props.acceleratorType,
+      initialInstanceCount: props.initialInstanceCount || 1,
+      initialVariantWeight: props.initialVariantWeight || 1.0,
+      instanceType: props.instanceType || InstanceType.T2_MEDIUM,
+      modelName: props.model.modelName,
+      variantName: props.variantName,
+    };
+  }
+
+  /**
+   * Get instance production variants associated with endpoint configuration.
+   */
+  public get instanceProductionVariants(): InstanceProductionVariant[] {
+    return Object.values(this._instanceProductionVariants);
+  }
```

Comment:
I don't understand the need for a getter and to differentiate between `instanceProductionVariants` and `_instanceProductionVariants`.

Reply:
The member `_instanceProductionVariants` is a map from variant name to variant (of type `{ [key: string]: InstanceProductionVariant; }`) to aid with lookup operations by variant name (and avoid variant name duplication) whereas this getter is exposing the variants as an array (for use-cases in which consuming code may wish to examine all configured variants).

Reply:
After thinking about this a bit more in light of my marking Endpoint-only methods as `@internal`, I've decided to make the following changes:

- I've renamed `_instanceProductionVariants` to `instanceProductionVariantsByName` to make its purpose as a map cleaner.
- The now-`@internal` methods are prefixed with `_` to obey linting rules.

The `@internal` getter is still useful so that `Endpoint` can itself expose a list of configured variants, but it need not be exposed to customers..